packages/itwinui-react/src/utils/components/ShadowRoot.tsx

```diff
@@ -127,5 +135,13 @@ function useShadowRoot(
     }
   }, [css]);
 
+  // Re-create stylesheet if the element is moved to a different window (by AppUI)
+  React.useEffect(() => {
+    window.addEventListener('appui:reparent', createStyleSheet);
+    return () => {
+      window.removeEventListener('appui:reparent', createStyleSheet);
+    };
+  }, [createStyleSheet, latestShadowRoot]);
```

Comment:
@GerardasB What do you think would be the best way to test this? It would be difficult to add a proper test for this in iTwinUI without manually recreating the custom event / reparenting logic.

I'm thinking we first manually verify that it works within the AppUI repo (by editing `@itwin/itwinui-react` in `node_modules`) and then after this is released, we can update the existing [AppUI test scenario](https://github.com/iTwin/appui/blob/6d28327aed77154e0814fcde526f78fee4c18cf2/e2e-tests/tests/popout-widget.test.ts#L207) to ensure the styles look fine without any special handling on top of `ProgressRadial`.

Reply:
Sounds good, I'll verify if this change works and let you know.

Reply:
I can confirm, this works correctly in the reparenting use-case (I made sure to remove the custom `useEffect` workaround https://github.com/iTwin/appui/blob/b1c1d0c95a58ebbf2985bb2e61ad01d639e7f795/apps/test-app/src/frontend/appui/frontstages/TestPopoutFrontstage.tsx#L90)

**However**, this change breaks the regular use-case, where no reparenting is done and the component is simply mounted in a new window. (`reparentPopoutWidgets=0` to configure in AppUI test-app: https://github.com/iTwin/appui/blob/b1c1d0c95a58ebbf2985bb2e61ad01d639e7f795/e2e-tests/tests/popout-widget.test.ts#L210)

This happens because `useEffect` that is scheduled in `useLatestRef(shadowRoot)` runs **after** the `createStyleSheet` called after `ReactDOM4.flushSync()`.

Reply:
Thanks for testing it. I believe I've fixed the regular popout case in my latest changes. I'm passing the already existing `shadow` into `createStyleSheet` (rather than relying on `useLatestRef`).

I also added a basic test for it, since this test does not require us to reimplement any reparenting logic.

Reply:
Looks good now!